### PR TITLE
Gpl 686 modify pacbio tags

### LINKED ISSUE #533: GPL-686 As a long read user I would like to be able to modify library tags without an RT request
**User story**
As a long read user I would like to be able to modify library tags without an RT request

**Who are the primary contacts for this story**
Craig

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] when I edit a library I should be able to change a tag

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

**Additional context**

### PR DESCRIPTION
**Pull request by BenTopping**
Closes #533 

Changes proposed in this pull request:

*You can now edit tags within the show details button on pacbio libraries
*
* ...

### CODE REVIEW

File: src/views/pacbio/PacbioLibraries.vue [hunk 1]
```diff
@@ -66,6 +66,10 @@
               :items= row.item.requests
               :fields="field_in_details"
               :filter="filter">
+                <template v-slot:cell(edit_tag)="row">
+                  <PacbioTagEditModal :tag="row.item" @alert="showAlert" @reloadPage="provider" >
```
Comment (harrietc52) [inline src/views/pacbio/PacbioLibraries.vue:70]: Ah haven't seen the `@reloadPage` before - that's nice!

File: src/views/pacbio/PacbioLibraries.vue [hunk 2]
```diff
@@ -118,7 +123,8 @@ export default {
   components: {
     Alert,
     PrinterModal,
-    PacbioLibraryEditModal
+    PacbioLibraryEditModal,
+    PacbioTagEditModal
```
Comment (harrietc52) [inline src/views/pacbio/PacbioLibraries.vue:127]: Neat 👍

File: src/components/pacbio/PacbioTagEditModal.vue [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  name: 'PacbioLibraryTagModal',
+  mixins: [ModalHelper],
+  data() {
+    return {
+      tags: [],
+      selectedSampleTagId: ""
+    }
+  },
+  props: {
+    tag: {
```
Comment (harrietc52) [inline src/components/pacbio/PacbioTagEditModal.vue:53]: Renaming this :)

File: src/components/pacbio/PacbioTagEditModal.vue [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  <div>
+    <b-button :id="'editTag'+this.tag.id" size="sm" @click="show" variant="outline-primary">Edit</b-button>
+
+    <b-modal
+      id="editTagModal"
+      ref="modal"
+      title="Edit Tag"      
+    >    
+
+      <b-form id="editTagForm" v-on:submit.prevent>
```
Comment (harrietc52) [inline src/components/pacbio/PacbioTagEditModal.vue:11]: What does submit.prevent do?

File: src/components/pacbio/PacbioTagEditModal.vue [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        <b-form-group id="input-group-1"
+                      label="Sample tag: "
+                      label-for="input-1">
+                      <span>{{ this.tag.sample_name }}</span>
```
Comment (harrietc52) [inline src/components/pacbio/PacbioTagEditModal.vue:15]: Dont think you need span and maybe the sample name could go along side Sample tag so its would read in one line

File: src/components/pacbio/PacbioTagEditModal.vue [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  },
+  props: {
+    tag: {
+      type: [Object]
+    }
+  },
+  methods: {
+    async update() {
+      try {
+        let payload = {selectedSampleTagId: this.selectedSampleTagId, request_library_id: this.tag.id }
```
Comment (harrietc52) [inline src/components/pacbio/PacbioTagEditModal.vue:60]: this.tag will be updated too from the rename :)

Comment (harrietc52) [reply src/components/pacbio/PacbioTagEditModal.vue:60]: just check over your spaces and formatting throughout

File: src/components/pacbio/PacbioTagEditModal.vue [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        await this.updateTag(payload)
+        this.alert('Tag updated', 'success')
+      } catch (err) {
```
Comment (harrietc52) [inline src/components/pacbio/PacbioTagEditModal.vue:63]: I'm not sure this ever gets hit? Does `updateTag` throw an exception that can be caught or do you have to check in the response and then handle the alert message accordingly?

File: src/components/pacbio/PacbioTagEditModal.vue [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        this.alert('Failed to update Tag. ' + err, 'danger')
+      }
+      this.hide()
+      this.$emit('reloadPage')
```
Comment (harrietc52) [inline src/components/pacbio/PacbioTagEditModal.vue:67]: Hm this makes me wonder that something isn't quite right if you have to manually reload the page - might be something in the VueX store asyn reloading. Will have a proper look after lunch

File: src/components/pacbio/PacbioTagEditModal.vue [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+    },
+    async provider() {
+      try{
+        await this.setTags()
+        this.tags = this.tractionTags
+          .filter(tag => tag.tag_set_id == 1)
```
Comment (harrietc52) [inline src/components/pacbio/PacbioTagEditModal.vue:73]: How come its tag_set_id==1? If its important probably good form to make it a constant, or add a clean comment etc

File: src/components/pacbio/PacbioTagEditModal.vue [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+          .map(tag => ({ text: tag.group_id, value: tag.id }))
+      } catch (error) {
```
Comment (harrietc52) [inline src/components/pacbio/PacbioTagEditModal.vue:75]: Again just want to check that an error is throw in setTags that could be caught here

**Review by harrietc52 — COMMENTED** [review]
Added a few comments, its really not far off though :) any questions let me know 

Also, I can't see how it wouldn't work if a library did happen to have more than one sample and tag, as the modal is for each row of the sample/tag lists, but maybe worth checking with a dummy bit of data to check. I know this won't be the case anyway but think the code could handle it if it was

File: src/components/pacbio/PacbioTagEditModal.vue (hunk 4 above)
Comment (BenTopping) [reply src/components/pacbio/PacbioTagEditModal.vue:11]: not needed in this instance, I will remove it, it is a vue method with stops a form being submitted

File: src/components/pacbio/PacbioTagEditModal.vue (hunk 5 above)
Comment (BenTopping) [reply src/components/pacbio/PacbioTagEditModal.vue:15]: OK, changed so it reads Sample tag: {sample_name}

File: src/components/pacbio/PacbioTagEditModal.vue (hunk 9 above)
Comment (BenTopping) [reply src/components/pacbio/PacbioTagEditModal.vue:73]: The tag_set_id == 1 is the pacbio tag set, other tag_sets such as ONT may have tag_set_id == 2, this is done so not every tag in traction is displayed in the drop down, only the pacbio ones

Comment (harrietc52) [reply src/components/pacbio/PacbioTagEditModal.vue:72]: Ok :) So maybe just adding a comment or making the 1 a constant with name PACBIO_TAG_SET to make it clear

**Review by harrietc52 — APPROVED** [review]

**Pull request merged**